src/box/index.cc

```diff
@@ -549,6 +549,28 @@ index_build(struct index *index, struct index *pk)
 	return 0;
 }
 
+bool index_tuple_is_excluded(struct index *index, struct tuple **old_tuple, struct tuple **new_tuple)
+{
+	if (*new_tuple == nullptr)
+		return false;
+	struct key_def* key_def = index->def->key_def;
+	struct key_part* parts = key_def->parts;
+	for (uint32_t i = 0; i < key_def->part_count; ++i) {
+		if (!parts[i].exclude_null)
+			continue;
+		const char* field = tuple_field(*new_tuple, parts[i].fieldno);
+		if (mp_typeof(*field) == MP_NIL) {
+			if (old_tuple && *old_tuple != nullptr) {
+				*new_tuple = nullptr;
+				return false;
+			} else {
```

Comment:
What if old_tuple was excluded? As I see in this case we are deleting it from tree.. It does not actually delete anything because we didn't insert it, but it looks buggy.